src/Symfony/Component/Yaml/Command/LintCommand.php

```diff
@@ -137,40 +143,75 @@ private function display(SymfonyStyle $io, array $files): int
                 return $this->displayTxt($io, $files);
             case 'json':
                 return $this->displayJson($io, $files);
+            case 'github':
+                return $this->displayTxt($io, $files, true);
             default:
                 throw new InvalidArgumentException(sprintf('The format "%s" is not supported.', $this->format));
         }
     }
 
-    private function displayTxt(SymfonyStyle $io, array $filesInfo): int
+    private function displayTxt(SymfonyStyle $io, array $filesInfo, bool $errorAsGithubAnnotations = false): int
     {
         $countFiles = \count($filesInfo);
         $erroredFiles = 0;
         $suggestTagOption = false;
 
+        $buffer = [];
         foreach ($filesInfo as $info) {
             if ($info['valid'] && $this->displayCorrectFiles) {
                 $io->comment('<info>OK</info>'.($info['file'] ? sprintf(' in %s', $info['file']) : ''));
             } elseif (!$info['valid']) {
                 ++$erroredFiles;
-                $io->text('<error> ERROR </error>'.($info['file'] ? sprintf(' in %s', $info['file']) : ''));
-                $io->text(sprintf('<error> >> %s</error>', $info['message']));
 
                 if (false !== strpos($info['message'], 'PARSE_CUSTOM_TAGS')) {
                     $suggestTagOption = true;
                 }
+
+                if ($errorAsGithubAnnotations) {
+                    $buffer[] = $this->formatErrorAsGithubAnnotation($info);
+                    continue;
+                }
+
+                $io->text('<error> ERROR </error>'.($info['file'] ? sprintf(' in %s', $info['file']) : ''));
+                $io->text(sprintf('<error> >> %s</error>', $info['message']));
             }
         }
 
         if (0 === $erroredFiles) {
             $io->success(sprintf('All %d YAML files contain valid syntax.', $countFiles));
         } else {
-            $io->warning(sprintf('%d YAML files have valid syntax and %d contain errors.%s', $countFiles - $erroredFiles, $erroredFiles, $suggestTagOption ? ' Use the --parse-tags option if you want parse custom tags.' : ''));
+            $message = sprintf('%d YAML files have valid syntax and %d contain errors.%s', $countFiles - $erroredFiles, $erroredFiles, $suggestTagOption ? ' Use the --parse-tags option if you want parse custom tags.' : '');
+
+            if ($errorAsGithubAnnotations) {
+                $io->writeln(sprintf('::warning::%s', $message));
+                $io->writeln($buffer);
```

Comment:
Outputting the summary warning first, so it appears first in the Github annotations on the PR checks

Reply:
if you remove the `continue` as decided in https://github.com/symfony/symfony/pull/38982/files#r517253530 this is not true anymore, output will be:
- IO errors
- Github Warning
- Github errors (buffered)
- IO warning. (if you also revert the else statement here)

Reply:
well, you would need to put the IO warning before the `if` rather than after.